src/shell/shell_misc.cpp

```diff
@@ -398,11 +398,15 @@ void DOS_Shell::ProcessCmdLineEnvVarStitution(char* line) {
 	std::smatch match;
 	/* Iterate over potential %var1%, %var2%, etc matches found in the text string */
 	while (std::regex_search(text, match, re)) {
-		std::string variable_name;
-		variable_name = match[1].str();
-		if (!variable_name.size()) {
+		// Get the first matching %'s position and length
+		const auto percent_pos = static_cast<std::string::size_type>(match.position(0));
+		const auto percent_len = static_cast<std::string::size_type>(match.length(0));
```

Comment:
Change `std::string::size_type` to `size_t`, please - it will be easier to read and understand. Fully expanded types exist mainly for deducting types in templated code.